src/forms/inputs/textarea.js

```diff
@@ -55,17 +55,17 @@ function Textarea (props) {
     ...rest
   } = omitLabelProps(props)
   return (
-    <LabeledField 
+    <LabeledField
       className={ classnames(className, { 'with-character-count': showCharacterCount }) }
       { ...props }
     >
-      { 
-        showCharacterCount &&
+      {
+        maxLength && showCharacterCount &&
```

Comment:
Because of the tricky `&&` thing, a `maxLength` of `0` would render "0" here.

Time to use some weird JS syntax :) https://stackoverflow.com/questions/784929/what-is-the-not-not-operator-in-javascript